---
uint8 XMLRPC_TYPE_INVALID = 0
uint8 XMLRPC_TYPE_BOOLEAN = 1
uint8 XMLRPC_TYPE_INT = 2
uint8 XMLRPC_TYPE_DOUBLE = 3
uint8 XMLRPC_TYPE_STRING = 4
uint8 XMLRPC_TYPE_DATETIME = 5
uint8 XMLRPC_TYPE_BASE64 = 6
uint8 XMLRPC_TYPE_ARRAY = 7
uint8 XMLRPC_TYPE_STRUCT = 8

string name           # ROS name of the parameter under which it can be accessed through rosparam
bool valid            # True, if the parameter value is accessible through rosparam
bool cached           # True, if the parameter service server provides the cached parameter value

uint8 xmlrpc_type     # XML-RPC type of the paramter's value
uint32 xmlrpc_size    # XML-RPC size for string, base64, array, and struct parameter values

string strong_type    # Human-readable strong type of the parameter's value
